src/core/thread/mesh_forwarder.cpp

```diff
@@ -458,6 +471,19 @@ Message *MeshForwarder::GetDirectTransmission(void)
 
         case Message::kTypeMacDataPoll:
             ExitNow();
+
+        case Message::kTypeSupervision:
+            if ((child = mNetif.GetChildSupervisor().GetDestination(*curMessage)) == NULL)
+            {
+                error = kThreadError_Drop;
+            }
+            else
+            {
+                mMacDest.mShortAddress = child->GetRloc16();
+                mMacDest.mLength = sizeof(mMacDest.mShortAddress);
+            }
+
+            break;
```

Comment:
remove this `case` block? 
supervision message is only for sleepy child, that is, indirect message.